### ISSUE TITLE
Question about openblas

### ISSUE THREAD

**Issue by linquanxu** [AlexeyAB/darknet#2007]
hi，AlexeyAB，
I want to use Openblas to speedup predict time and i add  cblas_sgemm to gemm.c. the predict image time  is about 4 seconds. 
![image](https://user-images.githubusercontent.com/16631303/49363781-8f97c280-f71d-11e8-8ad2-5736da0c91b3.png)
![image](https://user-images.githubusercontent.com/16631303/49363804-9c1c1b00-f71d-11e8-8c68-e7d611a99c25.png)

however,
when i used gemm_cpu instead of cblas_sgemm, predict time was 2.4 seconds.
![image](https://user-images.githubusercontent.com/16631303/49364123-6e83a180-f71e-11e8-8546-87e84f8f7766.png)
![image](https://user-images.githubusercontent.com/16631303/49364150-8d823380-f71e-11e8-973f-60eb24933df4.png)

I don't know why the speed of openblas is too slow.(do not use AVX ?)

**Comment by AlexeyAB** [comment]
@linquanxu Hi,

* How did you installed OpenBlas? 

* What version of OpenBlas do you use?

* Try to compile OpenBLAS: `make USE_OPENMP=1 NO_AVX=0` 

* What CPU do you use? https://github.com/xianyi/OpenBLAS#x86x86-64

* What GCC version do you use? `gcc --version`

* May be you compiled it without AVX2 & FMA or without multithreading? Does it use all CPU cores during detection on video or sequence of images?

I don't think that I made GEMM f32 implementation faster than OpenBlas, since I didn't implement GEMM for tiles for cache-friendly data loading. But also Intel MKL should be faster then OpenBlas.

**Comment by linquanxu** [comment]
> @linquanxu
> 
> Did you use `C:\Program Files (x86)\CMake\bin\cmake-gui.exe` to try to enable OpenMP and AVX/FMA for OpenBLAS?
> 
> Then enable OpenMP (`properties -> c/c++ -> language`) and AVX2 (`properties -> c/c++ -> code generation -> enable enhanced instruction set -> AVX2`)
> 
> If it doesn't help, try to ask it here: [xianyi/OpenBLAS#776](https://github.com/xianyi/OpenBLAS/issues/776)

  hi，@AlexeyAB, thank for help，I recompiled the OpenBLAS with SMY2 and minGW and test the predict time, Openblas is 2 times faster than used GEMM f32.


![image](https://user-images.githubusercontent.com/16631303/49653837-111a8800-fa71-11e8-94b2-69680bcff4de.png)

But it seems that  it have not used AVX.

**Comment by AlexeyAB** [comment]
@linquanxu Hi,

> But it seems that it have not used AVX.

Why do you think so?

What is the SMY2?
And what manual did you use for compiling using MinGW?

**Comment by linquanxu** [comment]
> @linquanxu Hi,
> 
> > But it seems that it have not used AVX.
> 
> Why do you think so?
> 
> What is the SMY2?
> And what manual did you use for compiling using MinGW?

when I built with GEMM f32, the print log was this,
![image](https://user-images.githubusercontent.com/16631303/49851526-53025000-fe1c-11e8-9305-e9c30e2de80f.png)

while I built with openblas, the printed  content of log did not contain AVX.
![image](https://user-images.githubusercontent.com/16631303/49851758-11be7000-fe1d-11e8-97a1-ef6be6b26759.png)
So, I guess, openblas may not have used AVX.


What is the SMY2?
I am sorry. Minimal GNU（POSIX）system on Windows(MSYS2) was been used, did not SMY2.
